spyder/utils/programs.py

```diff
@@ -462,6 +462,24 @@ def is_module_installed(module_name, version=None, installed_version=None,
             
             return check_version(actver, version, symb)
 
+def is_python_interpreter(filename):
+    """Evaluate wether a file is a python interpreter or not."""
+    real_filename = os.path.realpath(filename) # To follow symlink if existent
+    if not osp.isfile(real_filename) or encoding.is_text_file(real_filename):
+        return False
+    possible_names = ['python', 'python2', 'python2.7', 'python3', 'python3.4',
+                      'python3.5']

```

Comment:
@mariacamilaremolinagutierrez in order to test this, we need to break this small helper function

```python
def is_python_interpreter_valid_name(fname):
    """Check that the python interpreter file has a valid name."""
    pattern = r'python(?P<version>\d\.?\d*)?$'
    # etc......
```

That way we can test this method by itself to make sure it is grabbing the right stuff.